deployment/desktop/hosting-metadata.md

```diff
@@ -0,0 +1,198 @@
+
+1. Next, take the generated XML and the tarball of icons and add it to
+   the *repomd.xml* master document, so that GNOME Software
+   automatically downloads the content for searching:
+
+	```console
+$ modifyrepo_c					\
+	--no-compress				\
+	--simple-md-filenames			\
+	/tmp/asb-md/appstream.xml.gz		\
+	x86_64/repodata/
+$ modifyrepo_c					\
+	--no-compress				\
+	--simple-md-filenames			\
+	/tmp/asb-md/appstream-icons.tar.gz	\
+	x86_64/repodata/
+	```
+
+	Deploying this metadata allows GNOME Software to add the
+	application metadata the next time the repository is refreshed.
+
+## Hosting a DNF repository on Github
+
+While Github isn't ideal for hosting DNF repositories, this method currently works:
+
+1. Once you've created a local copy of your repository, create a new
+   project on Github. Then use the follow commands to import your
+   repository into Github.
+
+	```console
+$ cd ~/src/myrepository
+$ git init
+$ git add -A
+$ git commit -a -m "first commit"
+$ git remote add origin git@github.com:yourgitaccount/myrepo.git
+$ git push -u origin master
+	```
+
+1. Next, go to the Github web interface and browse down in the file
+   tree until you find the file called *repomd.xml*, and click on
+   it. You should now see a button in the Github interface called
+   *Raw*. Click that to get the raw version of the XML file.  In the
+   URL bar of your browser you should see a URL like this:
+
+	```console
+https://raw.githubusercontent.com/username/reponame/master/noarch/repodata/repomd.xml
+	```
+
+	Copy that URL, as you'll need it to create the *.repo* file that distributions and users use to reach your new repository.
+	
+1. Create a *.repo* file using this example as a template, and edit it to match your data:
+
+	```console
+[frobnicator]
+name=Frobnicator 3D generator
+baseurl=https://raw.githubusercontent.com/username/reponame/master/noarch
+gpgcheck=0
+enabled=1
+enabled_metadata=1
+	```
+
+	For more information on this format, consult the [repository
+    options in the DNF configuration documentation](http://dnf.readthedocs.io/en/latest/conf_ref.html#repo-options).
+	
+1. Copy this file to */etc/yum.repos.d* on your computer and load up
+   GNOME Software. Click on the *Updates* button in GNOME Software,
+   and then select the refresh button in the top left corner to ensure
+   your database is up to date.  You should then be able to search in
+   GNOME software and find your new application.
+
+## Generating Flatpak metadata
+
+The [Flatpak](https://flatpak.org) application packaging standard is another way to provide metadata for consumers. This guide assumes you are familiar with this standard, along with the [manifest and basic build operations required](http://docs.flatpak.org/en/latest/flatpak-builder.html).  You can find extensive information on building Flatpaks and on hosting and signing flatpak repostories [here](https://blogs.gnome.org/alexl/2017/02/10/maintaining-a-flatpak-repository/).
```

Comment:
> This guide assumes you are familiar with this standard, along with...

Please do remove that sentence, as well as the following one. I do not think it's relevant, but you can include those two links in a `reference` section.